### ISSUE TITLE
Clicking cancel crashes on MacOS

### ISSUE THREAD

**Issue by TheCleric** [ttytm/osdialog-zig#1]
I've added a simple file path picker in my zig app, and when I select a file all works great.

However if I hit cancel the app crashes with the following error:

```
thread 1187422 panic: reached unreachable code
/opt/homebrew/Cellar/zig/0.13.0/lib/zig/std/debug.zig:412:14: 0x102ac708b in assert (zig-fourier)
    if (!ok) unreachable; // assertion failure
             ^
/opt/homebrew/Cellar/zig/0.13.0/lib/zig/std/mem.zig:1017:23: 0x102afd40b in len__anon_9991 (zig-fourier)
                assert(value != null);
                      ^
/opt/homebrew/Cellar/zig/0.13.0/lib/zig/std/mem.zig:791:18: 0x102aec453 in span__anon_9722 (zig-fourier)
    const l = len(ptr);
                 ^
/Users/TheCleric/.cache/zig/p/12209d447a0347024cce59bb62e2012402c25ce6196ade3f284b1248a240fb1ad23e/src/bindings.zig:14:39: 0x102ad667f in toZString (zig-fourier)
 const z_string: [:0]u8 = std.mem.span(c_string);
                                      ^
/Users/TheCleric/.cache/zig/p/12209d447a0347024cce59bb62e2012402c25ce6196ade3f284b1248a240fb1ad23e/src/lib.zig:70:27: 0x102ab0c97 in path (zig-fourier)
 return bindings.toZString(allocator, bindings.osdialog_file(@intFromEnum(action), options.path, options.filename, null));
                          ^
/Users/TheCleric/Dev/personal/zig-fourier/src/main.zig:49:25: 0x102aaf9ab in main (zig-fourier)
            if (osd.path(allocator, .open, .{})) |path| {
                        ^
/opt/homebrew/Cellar/zig/0.13.0/lib/zig/std/start.zig:524:37: 0x102aaf5b3 in main (zig-fourier)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x19ac3c273 in ??? (???)
???:?:?: 0x0 in ??? (???)
```

**Comment by ttytm** [comment]
Thanks for reporting the issue! Gonna look into it tomorrow. Good time until then!

**Comment by ttytm** [comment]
Should be fixed in 8ea6ae8eb69b14610f15311de79dd68063321b7e.

Sorry about the inconvenience.
Let me know if you encounter other issues.

**Comment by TheCleric** [comment]
Thanks!
